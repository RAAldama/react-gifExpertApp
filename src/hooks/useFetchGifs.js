import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [data, setData] = useState({
        data: [], 
        loading: true
    });

    //Carga lo que está dentro según las condiciones, vacío es una vez cuando se carga 
    //el componente, 'category' se redibuja al cambiar category
    useEffect( () => {
        
        getGifs(category).then(imgs => {
            setData({
                data: imgs,
                loading: false
            });
        })

    }, [category]);

    return data
}