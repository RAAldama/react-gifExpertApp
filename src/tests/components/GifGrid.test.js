import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Tests on <GifGrid />', () => {

    const category = 'eevee'
    
    test('should show <GifGrid /> correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show items when images are loaded with useFetchGif', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/any/thing.jpg',
            title: 'Cualquier cosilla'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow( <GifGrid category={category} /> );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('#loading').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    
})
