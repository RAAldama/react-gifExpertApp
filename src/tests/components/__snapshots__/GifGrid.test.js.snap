// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid /> should show <GifGrid /> correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    eevee
  </h3>
  <p
    className="animate__animated animate__flash"
    id="loading"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests on <GifGrid /> should show items when images are loaded with useFetchGif 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    eevee
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosilla"
      url="https://localhost/any/thing.jpg"
    />
  </div>
</Fragment>
`;
